### ISSUE TITLE
Map Drawing: Refactor coordinate assigning code using Game State structs/functions

### ISSUE THREAD

**Issue by omorkved** [uchicago-cs/chiventure#131]
Refactor the code in coordinate.c and DFS.c to make use of the relevant functions provided by Game State and to no longer rely on dummy structs. (Especially accessing adjacent rooms within a room struct)

**Comment by omorkved** [comment]
Current structs that need to be changed
coordinate_t in coordinate.h
room_t in coordinate.h
coord_t in map.h (same as coordinate_t but with three fields. duplicate because came from another feature branch)
room_t in map.h (duplicate)

To deal with the room_t structs, I will delete these and use the game state structs instead. There might be a few changes to make in handling rooms within functions depending on how game state has written the current struct.

To deal with the coordinate structs: I will keep the coordinate_t in coordinate.h, as I feel like it makes the most sense to have the struct defined in this module. I will also have to make a decision about whether to support two dimensions (x, y) or three (x, y, z). It looks like Sam already has a system to handle visualization of multiple floors (by using key handling to switch between floors). The coordinate struct acts as the key to the hash table, and uthash works by literally comparing all of the bytes in a struct to see if two structs are identical, so adding a third int field into the struct should not affect how uthash stores and recieves keys. Therefore, I am currently planning to add in a z-direction. I will ask Game State and WDL to support this third direction, but even if they don't, at least we will have the infrastructure in place in case this is supported in the future. (If a third-direction/z-axis is not implemented in this sprint, then I will just set the z value to be 0 in coord_init() in coordinate.c for every coordinate as a temporary measure).

This task also involves switching over to gamestate functions, particularly a function for finding whether a direction exists in a room, and if so, what is room pointer matches that direction. (This will be used extensively in DFS.c)

**Comment by omorkved** [comment]
Ran into something unexpected -- game-state is using a char * for room IDs instead of integers. In DFS.c, we had originally been checking equality of room IDs to figure out whether an assigned coordinate was already assigned to the same room that we were checking (as opposed to attempting to double-assign a coordinate to a different room). However, I realized this could be solved by using strcmp() on the char * IDs so it's not a big deal. This should be fine as long as rooms are required to have _unique_ room IDs.

**Comment by omorkved** [comment]
Update: This code was successfully merged with the rest of the code in Sebastian's cli-integration branch and with our context branch, and has just been merged into master. We have not run into any other issues integrating with Game State.

One remaining task -- we are waiting on a function from Game State, find_room_from_dir() to make it into master before we use it (used in map.c and DFS.c). Replacing the dummy code and adding this in will be quite quick, so I am going to close this issue unless you find anything else missing here. @mpcsh

**Comment by mpcsh** [comment]
Issue Score: ✓+

Comments:
Great work!
